pyanaconda/modules/network/nm_client.py

```diff
@@ -55,23 +58,91 @@
 
 @contextmanager
 def nm_client_in_thread():
-    """Create NM Client with new GMainContext to be run in thread.
+    """Create temporary NMClient with new GMainContext to be run in thread.
 
-    Expected to be used only in installer environment for a few
-    one-shot isolated network configuration tasks.
-    Destroying of the created NM Client instance and release of
-    related resources is not implemented.
+    Passes a weak reference to the newly created NMClient.
+    The NMClient will be destroyed when leaving the context if there are no
+    other references preventing it.
 
     For more information see NetworkManager example examples/python/gi/gmaincontext.py
     """
     mainctx = create_new_context()
     mainctx.push_thread_default()
 
     try:
-        yield get_new_nm_client()
+        nm_client = get_new_nm_client()
     finally:
         mainctx.pop_thread_default()
 
+    # Pass a weak reference to the context so that it does not refer it
+    # when we try to destroy it on context exit.
+    nm_client_ref = weakref.ref(nm_client)
+    yield nm_client_ref
```

Comment:
You yield the reference to nm client after `mainctx.pop_thread_default()`, so the nm client will be used outside of the context of the current thread. That doesn't seem right.